channeldb/db.go

```diff
@@ -1410,9 +1410,12 @@ func (c *ChannelStateDB) FetchHistoricalChannel(outPoint *wire.OutPoint) (
 		}
 
 		channel, err = fetchOpenChannel(chanBucket, outPoint)
+		if err != nil {
```

Comment:
I don't think this check is needed, since on error, we already returned `err`.

Reply:
Agreed with Andras, or is the fear that we're overshadowing it?

Reply:
To avoid running into a nil pointer below when the channel is nil.

Reply:
Yes, it caused SIGSEGV.

Reply:
Ok I see now, `channel` was set to `nil` in `fetchOpenChannel`